### PR TITLE
test: adding user credentials to credential management system

### PR DESCRIPTION
**Pull request by hung-cybo**
## Why

- Adding user credentials to the credential management system

## What

- User credentials are added to the credential management system

## How to test

```
pnpm build
pnpm test:e2e
```

## Checklist

- [x] Read [CONTRIBUTING.md](https://github.com/kintone/cli-kintone/blob/main/CONTRIBUTING.md)
- [x] Updated documentation if it is required.
- [x] Added/updated tests if it is required. (or tested manually)
- [x] Passed `pnpm lint` and `pnpm test` on the root directory.

### CODE REVIEW

File: e2e-credentials-schema.json [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+                }
+              }
+            }
+          },
+          "userPermissions": {
+            "type": "array",
+            "items": {
+              "type": "object",
+              "properties": {
+                "key": { "type": "string" },
```
Comment (tasshi-me) [inline e2e-credentials-schema.json]: According to the App and types, we don't need `key` property.

```suggestion
```

File: features/utils/credentials.ts [hunk 2]
```diff
@@ -55,54 +104,40 @@ const e2eCredentialFilePath = path.join(
...
   const credentials = await loadFromKintone();
   if (isRunOnActions()) {
-    credentials.forEach((credential) => {
-      credential.apiTokens.forEach((apiToken) => {
-        if (apiToken.token.length > 0) {
-          core.setSecret(apiToken.token);
-        }
-      });
-    });
+    setSecret(credentials);
```
Comment (tasshi-me) [inline features/utils/credentials.ts:114]: Nice refactoring 👍

File: features/utils/credentials.ts [hunk 3]
```diff
@@ -129,57 +164,85 @@ const generateErrorMessages = (errors: ErrorObject[]): string[] => {
   });
 };
 
-const loadFromKintone: () => Promise<Credential[]> = async () => {
+const loadFromKintone: () => Promise<Credentials> = async () => {
   const kintoneBaseUrl = process.env.TEST_KINTONE_BASE_URL;
-  const kintoneAppId = process.env.TEST_KINTONE_CREDENTIAL_MANAGEMENT_APP_ID;
-  const kintoneApiToken =
+  const kintoneCMAAppId = process.env.TEST_KINTONE_CREDENTIAL_MANAGEMENT_APP_ID;
+  const kintoneCMAApiToken =
```
Comment (tasshi-me) [inline features/utils/credentials.ts]: Now CMA (Credential Management App) should be ACMA (App Credential Management App), as same as UCMA.
If the name ACMA and UCMA is not easy-to-read, I propose to use AppCMA and UserCMA.

File: features/common.feature [hunk 4]
```diff
@@ -1,25 +1,44 @@
...
+    When I run the command with args "record export --app 1 --base-url http://example.com --foo"
+    Then I should get the exit code is non-zero
+    And The output error message should match with the pattern: "Unknown argument: foo"
+
   Scenario: CliKintoneTest-2 Should return the error message when specifying unknown subcommand
     When I run the command with args "export"
     Then I should get the exit code is non-zero
     And The output error message should match with the pattern: "Unknown argument: export"
 
+  Scenario: CliKintoneTest-3 Should return the correct cli-kintone version when using `cli-kintone --version`
```
Comment (tasshi-me) [inline features/common.feature:13]: NOTE: moved from `version.feature`

**Review by tasshi-me — CHANGES_REQUESTED** [review]
I added two reviews!

File: features/utils/credentials.ts (hunk 3 above)
Comment (hung-cybo) [reply features/utils/credentials.ts]: Yes, I think AppCMA and UserCMA are more readable.

File: features/utils/world.ts [hunk 5]
```diff
@@ -32,7 +34,7 @@ export class OurWorld extends World {
     return this._credentials;
   }
 
-  public set credentials(value: Credential[]) {
+  public set credentials(value: Credentials) {
```
Comment (tasshi-me) [inline features/utils/world.ts]: It's better to add `init()` method and remove the setter for `credentials`.

We don't expect `credentials` to be changed while testing.
Deleting the setter lets developers know they should not update credentials excepting via `init()`

NOTE: We also want to forbid updating `workingDir`, but now it is updated on `Before("@isolated")`.

**Review by tasshi-me — APPROVED** [review]
LGTM!!

**Pull request merged**